### PR TITLE
Idiomatic adjoints for bloqs

### PR DESCRIPTION
**Pull request by mpharrigan**
Following #565 

This PR updates bloq to use the new adjoint protocol

 - basic_gates learn their adjoints
 - `adjoint: bool` flag removed from bloqs that do not have special adjoint behavior.
 - flag renamed for those that do

### CODE REVIEW

File: qualtran/Adjoint.ipynb [hunk 1]
```diff
@@ -26,7 +26,9 @@
     "from qualtran.drawing import show_bloq\n",
     "from qualtran.bloqs.basic_gates import ZeroState\n",
     "\n",
+    "print(ZeroState())\n",
```
Comment (fdmalone) [inline qualtran/Adjoint.ipynb:29]: remove?

File: qualtran/_infra/gate_with_registers.py [hunk 2]
```diff
@@ -309,17 +309,21 @@ def decompose_from_registers(
     def _decompose_with_context_(
         self, qubits: Sequence[cirq.Qid], context: Optional[cirq.DecompositionContext] = None
     ) -> cirq.OP_TREE:
+        from qualtran.cirq_interop._cirq_to_bloq import InteropQubitManager
```
Comment (fdmalone) [inline qualtran/_infra/gate_with_registers.py:312]: Why not top level import?

**Review by fdmalone — APPROVED** [review]
LGTM. Mostly using the new adjoing + sneaking in a T gate.

File: qualtran/_infra/gate_with_registers.py (hunk 2 above)
Comment (mpharrigan) [reply qualtran/_infra/gate_with_registers.py:312]: this is from #542

**Comment by mpharrigan** [comment]
I reverted the changes to some of the comparison gates, tracking in https://github.com/quantumlib/Qualtran/issues/580

File: qualtran/Adjoint.ipynb (hunk 1 above)
Comment (mpharrigan) [reply qualtran/Adjoint.ipynb:29]: this is on purpose to show that it turns into "ZeroEffect"

**Comment by mpharrigan** [comment]
Why was this able to be merged even though the tests were failing?

**Comment by tanujkhattar** [comment]
That's because we didn't update the branch protection rules after adding new github actions (for notebooks & pytest-dev-tools) so these tests were not marked as "required". I've updated the branch protections rules so that all the 6 CI actions are now required and this should not happen again.

**Pull request merged**